src/org/ohdsi/usagi/ui/Global.java

```diff
@@ -19,21 +19,9 @@
 
 import org.ohdsi.usagi.BerkeleyDbEngine;
 import org.ohdsi.usagi.UsagiSearchEngine;
-import org.ohdsi.usagi.ui.actions.AboutAction;
-import org.ohdsi.usagi.ui.actions.ApplyPreviousMappingAction;
-import org.ohdsi.usagi.ui.actions.ApproveAction;
-import org.ohdsi.usagi.ui.actions.ApproveAllAction;
-import org.ohdsi.usagi.ui.actions.ClearAllAction;
-import org.ohdsi.usagi.ui.actions.ConceptInformationAction;
-import org.ohdsi.usagi.ui.actions.ExitAction;
-import org.ohdsi.usagi.ui.actions.ExportForReviewAction;
-import org.ohdsi.usagi.ui.actions.ExportSourceToConceptMapAction;
-import org.ohdsi.usagi.ui.actions.ImportAction;
-import org.ohdsi.usagi.ui.actions.OpenAction;
-import org.ohdsi.usagi.ui.actions.RebuildIndexAction;
-import org.ohdsi.usagi.ui.actions.SaveAction;
-import org.ohdsi.usagi.ui.actions.SaveAsAction;
-import org.ohdsi.usagi.ui.actions.ShowStatsAction;
+import org.ohdsi.usagi.ui.actions.*;
+
+import java.util.Vector;
 
 public class Global {
```

Comment:
General comment on Java architecture: it is generally considered preferable to pass around references to an `Application` class or specialised `Service` classes to get a hold of application-level configuration. This avoids putting unrelated variables from different system parts (logic, UI, I/O) in a single place, and it gives you a central place to make updates on any of the variables.

One option is to start using dependency injection (e.g. [HK2](https://javaee.github.io/hk2/introduction.html) or [Dagger](https://github.com/google/dagger) and dynamically inject application and services into your classes.

However, a more straightforward option is to create an `UsagiApplication` class with references to the separate services (e.g. `getAlertService()`, `getActionService()`). You can either pass that UsagiApplication around or make it a singleton (e.g., `INSTANCE = new UsagiApplication(); UsagiApplication.get() { return INSTANCE; }`)

Reply:
Another note: if you expect that an alternative implementation will be used at a later time, you can make a service an interface and return the implementation of that interface from the application. That way, you can switch out implementations based on the current application settings.

Reply:
Will park this for now, as this requires quite a refactoring of the application and in EHDEN we are considering developing a new application that replaces some of Usagi's functionalities